tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -162,78 +162,159 @@ int main(int argc, char **argv)
         };
     };
 
-    /* s2n_get_and_validate_negotiated_signature_scheme */
+    /* s2n_signature_algorithm_recv */
     {
-        struct s2n_config *config = s2n_config_new();
+        struct s2n_security_policy test_security_policy = *s2n_fetch_default_config()->security_policy;
+        test_security_policy.signature_preferences = &test_preferences;
 
-        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
-        s2n_connection_set_config(conn, config);
+        /* Test: successfully choose valid server signature */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            conn->security_policy_override = &test_security_policy;
 
-        const struct s2n_security_policy *security_policy = NULL;
-        EXPECT_SUCCESS(s2n_connection_get_security_policy(conn, &security_policy));
-        EXPECT_NOT_NULL(security_policy);
+            DEFER_CLEANUP(struct s2n_stuffer input = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&input, 0));
+            s2n_stuffer_write_uint16(&input, s2n_rsa_pkcs1_sha256.iana_value);
 
-        struct s2n_security_policy test_security_policy = {
-            .minimum_protocol_version = security_policy->minimum_protocol_version,
-            .cipher_preferences = security_policy->cipher_preferences,
-            .kem_preferences = security_policy->kem_preferences,
-            .signature_preferences = &test_preferences,
-            .ecc_preferences = security_policy->ecc_preferences,
+            EXPECT_OK(s2n_signature_algorithm_recv(conn, &input));
+            EXPECT_EQUAL(conn->handshake_params.server_cert_sig_scheme, &s2n_rsa_pkcs1_sha256);
         };
 
-        config->security_policy = &test_security_policy;
+        /* Test: successfully choose valid client signature */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            conn->security_policy_override = &test_security_policy;
+            /* Unlike clients, servers begin with actual_protocol_version unset */
```

Comment:
Do you mean set?

Reply:
Nope, unset: https://github.com/aws/s2n-tls/blob/main/tls/s2n_connection.c#L575 Clients guess at the actual_protocol_versions, servers don't.

Reply:
I don't think it's a helpful comment. I just deleted the comment and set the protocol version on both the client and server tests to avoid confusion.